frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/navigation/CheckoutPreviousAction.java

```diff
@@ -0,0 +1,28 @@
+package com.virtuslab.gitmachete.frontend.actions.navigation;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import io.vavr.control.Option;
+import lombok.experimental.ExtensionMethod;
+import lombok.val;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.frontend.actions.base.BaseCheckoutAction;
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeySelectedBranchName;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+
+@ExtensionMethod(GitMacheteBundle.class)
+public class CheckoutPreviousAction extends BaseCheckoutAction
+    implements
+      IExpectsKeySelectedBranchName {
+
+  @Override
+  protected @Nullable String getTargetBranchName(AnActionEvent anActionEvent) {
+    val currentBranchName = getCurrentBranchNameIfManaged(anActionEvent);
+    return Option
+        .ofOptional(deriveLayoutBranches(anActionEvent)
+            .takeWhile(branchName -> !branchName.equals(currentBranchName))
+            .reduce((first, second) -> second))
```

Comment:
Isn't there some method like `last()`? 🤔

Reply:
no, unfortunately. But this is going to get removed when the get-next-prev-parent gets merged.

Reply:
Yes, or even better, you can make this PR based on the get-next-prev-parent PR, for example using Git Machete 😁

Reply:
Yes! exactly